results/js/Smart.TimeResultFormatting.js

```diff
@@ -0,0 +1,93 @@
+
+function getBestTimes(table) {
+  var numColumns = table.rows[0].cells.length;
+  var best_times = Array(numColumns).fill(999999.99);
+  for (row = 1; row < table.rows.length; row++) {
+    var cells = table.rows[row].cells;
+    for (column = 1; column < cells.length; column++) {
+      var bestSoFar = best_times[column];
+      var divElement = findChildWithClassNames(cells[column], ['search_time', 'search_time_best']);
+      var currentValue = parseFloat(divElement.innerText);
+      if (currentValue < bestSoFar && currentValue > 0.0) {
+         best_times[column] = currentValue;
+      }
+    }
+  }
+  return best_times;
+}
+
+function heatMapGray(table, cutOffPercent) {
+  heatMap(table, 0, 0, cutOffPercent, 100, 95, 30);
+}
+
+function heatMapGreen(table, cutOffPercent) {
+  heatMap(table, 128, 75, cutOffPercent, 100, 95, 30);
+}
+
+function heatMapRed(table, cutOffPercent) {
+  heatMap(table, 0, 75, cutOffPercent, 100, 95, 30);
+}
+
+function heatMapBlue(table, cutOffPercent) {
+  heatMap(table, 240, 75, cutOffPercent, 100, 95, 30);
+}
+
+function heatMap(table, hue, saturation, cutOffPercent, cutOffLuminance, worstTimeLuminance, bestTimeLuminance) {
+  var luminanceRange = Math.abs(worstTimeLuminance - bestTimeLuminance);
+  var best_times = getBestTimes(table);
+  for (row = 1; row < table.rows.length; row++) {
```

Comment:
missing declaration `for (let row = 1; etc.....` same for coluumns

Reply:
added declaration, and got rid of the vars.